keras_rcnn/models/_rcnn.py

```diff
@@ -12,78 +12,90 @@
-
-        output_features = keras.layers.MaxPooling2D(strides=(2, 2), name='block1_pool')(output_features)
-
-        output_features = keras.layers.Conv2D(128, name='block2_conv1', **options)(output_features)
-        output_features = keras.layers.Conv2D(128, name='block2_conv2', **options)(output_features)
+        inputs = [
+            target_bounding_boxes,
+            target_categories,
+            target_image,
+            target_masks,
+            target_metadata
+        ]
 
-        output_features = keras.layers.MaxPooling2D(strides=(2, 2), name='block2_pool')(output_features)
+        output_features = keras.layers.Conv2D(64, **options)(target_image)
 
-        output_features = keras.layers.Conv2D(256, name='block3_conv1', **options)(output_features)
-        output_features = keras.layers.Conv2D(256, name='block3_conv2', **options)(output_features)
-        output_features = keras.layers.Conv2D(256, name='block3_conv3', **options)(output_features)
-        output_features = keras.layers.Conv2D(256, name='block3_conv4', **options)(output_features)
+        convolution_3x3 = keras.layers.Conv2D(64, **options)(output_features)
 
-        output_features = keras.layers.MaxPooling2D(strides=(2, 2), name='block3_pool')(output_features)
+        output_deltas = keras.layers.Conv2D(9 * 4, (1, 1), activation="linear", kernel_initializer="zero", name="deltas")(convolution_3x3)
 
-        output_features = keras.layers.Conv2D(512, name='block4_conv1', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block4_conv2', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block4_conv3', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block4_conv4', **options)(output_features)
+        output_scores = keras.layers.Conv2D(9 * 1, (1, 1), activation="sigmoid", kernel_initializer="uniform", name="scores")(convolution_3x3)
 
-        output_features = keras.layers.MaxPooling2D(strides=(2, 2), name='block4_pool')(output_features)
+        target_anchors, target_proposal_categories, target_proposal_bounding_boxes = keras_rcnn.layers.AnchorTarget()([output_scores, target_bounding_boxes, target_metadata])
 
-        output_features = keras.layers.Conv2D(512, name='block5_conv1', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block5_conv2', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block5_conv3', **options)(output_features)
+        output_deltas, output_scores = keras_rcnn.layers.RPN()([target_proposal_bounding_boxes, target_proposal_categories, output_deltas, output_scores])
 
-        convolution_3x3 = keras.layers.Conv2D(512, name="convolution_3x3", **options)(output_features)
+        output_proposal_categories = keras_rcnn.layers.ObjectProposal()([target_metadata, output_deltas, output_scores, target_anchors])
```

Comment:
reorder: target anchors first, metadata last

Reply:
Also I was confused at first what output proposal categories was (at first I thought they were something to do with classes but they don't)

Reply:
What does it return? The proposed bounding boxes? Objectness scores? Both?

Reply:
no, just proposed bounding boxes